plugin/trino-singlestore/src/main/java/io/trino/plugin/singlestore/SingleStoreClient.java

```diff
@@ -708,21 +709,39 @@ private static LongReadFunction singleStoreTimeReadFunction(TimeType timeType)
     {
         requireNonNull(timeType, "timeType is null");
         checkArgument(timeType.getPrecision() <= 9, "Unsupported type precision: %s", timeType);
-        return (resultSet, columnIndex) -> {
-            // SingleStore JDBC driver wraps time to be within LocalTime range, which results in values which differ from what is stored, so we verify them
-            String timeString = resultSet.getString(columnIndex);
-            try {
-                long nanosOfDay = LocalTime.from(ISO_LOCAL_TIME.parse(timeString)).toNanoOfDay();
-                verify(nanosOfDay < NANOSECONDS_PER_DAY, "Invalid value of nanosOfDay: %s", nanosOfDay);
-                long picosOfDay = nanosOfDay * PICOSECONDS_PER_NANOSECOND;
-                long rounded = round(picosOfDay, 12 - timeType.getPrecision());
-                if (rounded == PICOSECONDS_PER_DAY) {
-                    rounded = 0;
+        return new LongReadFunction()
+        {
+            @Override
+            public boolean isNull(ResultSet resultSet, int columnIndex)
+                    throws SQLException
+            {
+                String timeString = resultSet.getString(columnIndex);
+                try {
+                    return resultSet.wasNull();
+                }
+                catch (SQLException e) {
+                    throw new IllegalStateException(format("Supported Trino TIME type range is between 00:00:00 and 23:59:59.999999 but got %s", timeString), e);
```

Comment:
Do we really need this catch, or adding just in case? I'm asking this question because we don't have a catch in `SingleStoreClient.checkNullUsingBytes` method. 

SingleStore JDBC driver doesn't throw SQLException in `wasNull` method as far as I confirmed the code.

Reply:
I got what you ment, thanks